src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,70 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 500;
+
+    private final int[] array;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    public MergeSortAction(int[] array, int start, int end) {
+        this.array = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if (end - start <= THRESHOLD) {
+            sequentialSort(array, start, end);
+        } else {
+            int mid = (start + end) / 2;
+
+            MergeSortAction leftTask = new MergeSortAction(array, start, mid);
+            MergeSortAction rightTask = new MergeSortAction(array, mid, end);
+
+            invokeAll(leftTask, rightTask);
+
+            merge(array, start, mid, end);
+        }
+    }
+
+    private void sequentialSort(int[] array, int start, int end) {
+        for (int i = start; i < end; i++) {
+            for (int j = i + 1; j < end; j++) {
+                if (array[i] > array[j]) {
+                    int temp = array[i];
+                    array[i] = array[j];
+                    array[j] = temp;
+                }
+            }
+        }
+    }
+
+    private void merge(int[] array, int start, int mid, int end) {
```

Comment:
The merge method does not handle the case where the input subarrays are already sorted. This could lead to unnecessary copying of elements. Consider checking if the last element of the left subarray is less than or equal to the first element of the right subarray before merging.